src/ArborX_HDBSCAN.hpp

```diff
@@ -12,31 +12,44 @@
 #define ARBORX_HDBSCAN_HPP
 
 #include <ArborX_Dendrogram.hpp>
+#include <ArborX_DetailsKokkosExtScopedProfileRegion.hpp>
 #include <ArborX_MinimumSpanningTree.hpp>
 
 namespace ArborX::Experimental
 {
 
 template <typename ExecutionSpace, typename Primitives>
 auto hdbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
-             int core_min_size)
+             int core_min_size,
+             DendrogramImplementation dendrogram_impl =
+                 DendrogramImplementation::DEFAULT)
 {
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN");
+  KokkosExt::ScopedProfileRegion guard("ArborX::HDBSCAN");
 
   using MemorySpace = typename Primitives::memory_space;
 
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::mst");
-  Details::MinimumSpanningTree<MemorySpace> mst(exec_space, primitives,
-                                                core_min_size);
-  Kokkos::Profiling::popRegion();
+  if (dendrogram_impl == DendrogramImplementation::DEFAULT)
+    dendrogram_impl = DendrogramImplementation::BORUVKA;
 
-  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN::dendrogram");
-  Dendrogram<MemorySpace> dendrogram(exec_space, mst.edges);
-  Kokkos::Profiling::popRegion();
+  if (dendrogram_impl != DendrogramImplementation::BORUVKA)
```

Comment:
I think it's clearer to avoid the negation if possible and rather have the Boruvka implementation first.